subsys/bluetooth/controller/Kconfig

```diff
@@ -29,4 +28,23 @@ choice BLE_CONTROLLER_VARIANT
 
 endchoice
 
+comment "Thread safety behavior of BLE controller APIs"
+
+config BLECTLR_THREADSAFETY
+	bool "Enable API blocking for thread safety"
+	depends on MULTITHREADING
+	default y
+	help
+		Public API calls such as Crypto and HCI APIs will block
+		if called by multiple threads simultaneously. Only one thread
+		can access the APIs at a time.
+config BLECTLR_THREADSAFETY_LOCK_TIMEOUT_MSEC
```

Comment:
This cannot be configured globally. Only user of the API knows that he can or cannot wait. And there might be one thread which cannot wait and one which can.

Reply:
We need a solution for ensuring cross-API thread safety for the ble_controller. The best we could see (for now) was to let the user define it's worst-case timeout or remove the lock completely if he is sure there are no competing threads accessing the APIs. Do you have a suggestion for how to achieve this?

Reply:
Ideally the caller of the functions which are taking the lock should specify the timeout. I am aware that would require API changes in several places and will be not easy.

However introducing this option as-is leads to issues. Many API calls which waited forever for the lock could now block and then return an error which is not expected by the existing callers. IMHO the clock control is most affected here (note that our code mostly do not check the value returned by clock_control_on()).

@nvlsianpu: Could you check how this could affect flash?

Reply:
This configuration item has now been removed.